File: JavaScript/프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    const sObj = {}
    const answer = []
    s.split('').forEach((char, i) => {
        sObj.hasOwnProperty(char) ? answer.push(i - sObj[char]) : answer.push(-1)
        sObj[char] = i
    })
    
    return answer
}

File: Python3/프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
from collections import defaultdict

def solution(s):
    sDict = defaultdict(str)
    answer = []
    
    for i, char in enumerate(s) :
        if char in sDict:
            answer.append(i - sDict[char])
        else:
            answer.append(-1)
        sDict[char] = i
        
    return answer